src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,65 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAX_NUMBER = 10;
+    private static final int FIGURE_COUNT = 5;
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int index = random.nextInt(FIGURE_COUNT);
+
+        switch (index) {
+            case 0:
+                return getRandomCircle();
+            case 1:
+                return getRandomSquare();
+            case 2:
+                return getRandomRectangle();
+            case 3:
+                return getRandomRightTriangle();
+            default:
+                return getRandomIsoscelesTrapezoid();
+        }
+    }
+
+    public Figure getDefaultFigure() {
+        return new Circle(Color.WHITE, MAX_NUMBER);
+    }
+
+    public Circle getRandomCircle() {
```

Comment:
We should only have two public methods, the others should all be private:
```suggestion
    private Circle getRandomCircle() {
```